chill-scala/src/test/scala/com/twitter/chill/ClosureCleanerSpec.scala

```diff
@@ -16,59 +16,227 @@ limitations under the License.
 
 package com.twitter.chill
 
+import com.twitter.chill.ClosureCleaner.{deserialize, serialize}
 import org.scalatest._
+import org.scalacheck.Arbitrary
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks._
+
+import scala.reflect.ClassTag
+import scala.util.Try
+
+object ClosureCleanerSpec {
+  // scala 2.12.x has better Java 8 interop. Lambdas are serializable and implemented via
+  // invokeDynamic and the use of LambdaMetaFactory.
+  val supportsSerializedLambda: Boolean = scala.util.Properties.versionString.contains("2.12")
+}
 
 class ClosureCleanerSpec extends WordSpec with Matchers {
-  def debug(x: AnyRef) {
-    println(x.getClass)
-    println(x.getClass.getDeclaredFields.map { _.toString }.mkString("  "))
-  }
+  private val someSerializableValue = 1
+  private def someSerializableMethod() = 1
+
   "ClosureCleaner" should {
     "clean normal objects" in {
       val myList = List(1, 2, 3)
-      ClosureCleaner(myList)
-      myList should equal(List(1, 2, 3))
+      serializable(myList, before = true, after = true)
+
+      case class Test(x: Int)
+      val t = Test(3)
+      serializable(t, before = false, after = true)
     }
-    "clean actual closures" in {
-      val myFun = { x: Int =>
-        x * 2
+
+    "clean basic serializable closures" in {
+      val localVal = 1
+      val closure1 = (_: Int) => localVal
+      val closure2 = (x: Int) => x + localVal
+
+      serializableFn(closure1, before = true, after = true)
+      serializableFn(closure2, before = true, after = true)
+    }
+
+    "clean basic closures (non LMF)" in {
+      assume(!ClosureCleanerSpec.supportsSerializedLambda)
```

Comment:
I don't know what this assume construct does. I think it is the first use in these tests. Is it bailing out of the test if that is not true?

Reply:
due to the different closures implementations between 2.11 (anonymous class) and 2.12 (java.lang.invoke.LambdaMetaFactory) use cases in 2.11 that require closure cleaning in 2.12 is not needed. With the usage of `assume` we can actually "CANCEL" (avoids) the test execution if the predicate is true.